src/pages/AddEditTask.tsx

```diff
@@ -40,6 +42,41 @@ const AddTask = (props: AddTaskProps) => {
         ];
     });
 
+    // if the task already exists, populate the fields
+    useEffect(() => {
+        console.log(params)
+        async function fillBoxes(task_id: number) {
+            const task = await getTask(task_id);
+            setTitle(task.title);
+            setDueDate(task.due_date);
+            setDescription(task.description);
+            setDone(task.completed);
+            setColors(colors.map((color) => {
+                if (color.color === task.color) {
+                    return { color: color.color, selected: true };
+                }
+                return { color: color.color, selected: false };
+            }));
+            
+            const room = await getRoom(task.room_id);
+            console.log(room)
+            setSelectedRoom(room);
+
+            const tags_of_task = await getTagsOfTask(task_id);
+            const tags = await Promise.all(tags_of_task.map(async (tag_of_task) => {
+                return await getTag(tag_of_task.tag_name);
+            }));
+            setSelectedTags(tags);
+        }
+
+        if (params.taskid) {
+            fillBoxes(+params.taskid);
+        }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
```

Comment:
```suggestion
    }, [colors, params])
```
Throw `colors` and `params` into the dependency array. That way, React knows to re-run this useEffect if either of them changes. This kind of thing is what makes React able to know what to update in real-time.

Reply:
when I did this it freaked out for some reason and wouldn't let me edit the fields... not sure why

Reply:
working on it